app/models/state.rb

```diff
@@ -22,11 +22,29 @@ def self.publish(content_item)
     change_state(content_item, name: "published")
   end
 
-  def self.unpublish(content_item)
+  def self.unpublish(content_item, type:, explanation: nil, alternative_path: nil)
     change_state(content_item, name: "unpublished")
 
-    Unpublishing.create!(
-      content_item: content_item,
+    unpublishing = Unpublishing.find_by(content_item: content_item)
+
+    if unpublishing.present?
+      unpublishing.update_attributes(
+        type: type,
+        explanation: explanation,
+        alternative_path: alternative_path,
+      )
+    else
+      Unpublishing.create!(
+        content_item: content_item,
+        type: type,
+        explanation: explanation,
+        alternative_path: alternative_path,
+      )
+    end
+  end
+
+  def self.substitute(content_item)
+    unpublish(content_item,
       type: "substitute",
       explanation: "Automatically unpublished to make way for another content item",
     )
```

Comment:
What do you think about maybe making this `#unpublish_as_substitute` (or something better named) to distinguish it from those methods that relate directly with the `state.name`? Maybe I'm just being a pedant, though.